packages/blade/src/components/BottomSheet/BottomSheet.tsx

```diff
@@ -0,0 +1,413 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable react/jsx-no-useless-fragment */
+/* eslint-disable consistent-return */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import React from 'react';
+import styled from 'styled-components';
+import { rubberbandIfOutOfBounds, useDrag } from '@use-gesture/react';
+import usePresence from 'use-presence';
+import { BottomSheetGrabHandle, BottomSheetHeader } from './BottomSheetHeader';
+import { BottomSheetBody } from './BottomSheetBody';
+import type { SnapPoints } from './utils';
+import { computeMaxContent, computeSnapPointBounds } from './utils';
+import { BottomSheetBackdrop } from './BottomSheetBackdrop';
+import { BottomSheetContext, useDropdownBottomSheetContext } from './BottomSheetContext';
+import { ComponentIds } from './componentIds';
+import { BottomSheetCloseButton } from './BottomSheetCloseButton';
+import BaseBox from '~components/Box/BaseBox';
+import { makeMotionTime, makeSpace } from '~utils';
+
+import { useScrollLock } from '~src/hooks/useScrollLock';
+import { useWindowSize } from '~src/hooks/useWindowSize';
+import { useIsomorphicLayoutEffect } from '~src/hooks/useIsomorphicLayoutEffect';
+import { assignWithoutSideEffects } from '~src/utils/assignWithoutSideEffects';
+import { useTheme } from '~components/BladeProvider';
+
+type BottomSheetProps = {
+  isOpen?: boolean;
+  onDismiss?: () => void;
+  children: React.ReactNode;
+  initialFocusRef?: React.MutableRefObject<any>;
+  snapPoints: SnapPoints;
+};
+
+export const BOTTOM_SHEET_EASING = 'cubic-bezier(.15,0,.24,.97)';
```

Comment:
Because we don't use spring libraries to derive the motion, this is the easing token which we found by trial and error which looks the most smoothest. 

Our motion tokens easing looks jittery because they don't give us a smooth curve.

Reply:
@anuraghazra please do track adding this.

Reply:
Will track this, but IMO we shouldn't add this to tokens, this easing is quite specific to bottomsheet. Not sure if we should generalize this.